core/src/test/java/org/hibernate/ogm/sessionfactory/SessionFactoryConfig.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.sessionfactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.ogm.cfg.OgmConfiguration;
+import org.hibernate.ogm.utils.TestHelper;
+
+/**
+ * Builder class to create a test configuration
+ * to use within {@link SessionFactoryInlineTest} inline helper.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class SessionFactoryConfig {
+
+	// input
+	private final Class<?>[] entities;
+	private final Map<String, Object> customProperties = new HashMap<>();
+
+	// output
+	private Configuration configuration;
+	private StandardServiceRegistry serviceRegistry;
+
+	private SessionFactoryConfig(Class<?>[] entities) {
+		this.entities = entities;
+	}
+
+	public static SessionFactoryConfig entities(Class... entities) {
+		return new SessionFactoryConfig( entities );
+	}
+
+	public SessionFactoryConfig property(String key, Object value) {
+		customProperties.put( key, value );
+		return this;
+	}
+
+	public Configuration getConfiguration() {
+		return configuration;
+	}
+
+	public StandardServiceRegistry getServiceRegistry() {
+		return serviceRegistry;
+	}
+
+	public SessionFactoryConfig build() {
```

Comment:
And the `build()` method would simply have returned a `SessionFactory`.

You would avoid having 2 getters and a state that is not very clear.

Reply:
Good question again. But the creation of SesssionFactory is a critical moment and it should be performed by the `SessionFactoryInlineTest`. In particular an exception could be cached or not, it depends by the test expectations.

Reply:
Sure, you can just call the `build()` method at the right time in the test.

The fact is that you don't really need the config. And your building pattern for the config is not very satisfactory.

Reply:
Sorry @gsmet I got it now what you mean. It is definitely a good idea.